test/e2e/performanceprofile/functests/0_config/config.go

```diff
@@ -158,9 +164,18 @@ func externalPerformanceProfile(performanceManifest string) (*performancev2.Perf
 	return profile, nil
 }
 
-func testProfile() *performancev2.PerformanceProfile {
-	reserved := performancev2.CPUSet("0")
-	isolated := performancev2.CPUSet("1-3")
+func testProfile() (*performancev2.PerformanceProfile, error) {
+	//Get cpus capacity on a node; typically it is same for all nodes
+	workerNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+	Expect(err).ToNot(HaveOccurred())
+	capacityCPU, _ := workerNodes[0].Status.Capacity.Cpu().AsInt64()
+	//consider cpu siblings - avoid noisy neighbor case
+	if capacityCPU < 4 {
+		//should never reach but still
+		return nil, fmt.Errorf("insufficient cpus on a node to create a valid performanceprofile for the tests, found %d cpus, expects at least 4", capacityCPU)
+	}
+	reserved := performancev2.CPUSet(fmt.Sprintf("0,%d", capacityCPU/2))
```

Comment:
I strongly recommend to use CPUSets and do logic with that instead of using just numbers